lib/libspl/list.c

```diff
@@ -62,10 +62,9 @@ void
 list_create(list_t *list, size_t size, size_t offset)
 {
 	ASSERT(list);
-	ASSERT(size > 0);
-	ASSERT(size >= offset + sizeof (list_node_t));
```

Comment:
I would keep these and below assertions to keep future code compatible with Illumos or whatever other port that may still use the size until we decide to remove it completely.

Reply:
Okay, I've added those assertions back in.